Landry333/Big-Owl
# #273 bug fix contacts black screen

### Related Issues
Closes #273

### Description
Contacts require a database call to get in Android, so I followed Along with Android's explanation of how to get a Contacts list and used a `LoaderManager` to get the contacts, which gets them asyncly. Furthermore, I added a max of 50 results upon contacts loading, to get more specific result, the user can search the first name, last name, or number of the contacts.

Other improvements added was the use of the number formatter, which will make the results filter out numbers that cant exist. I also added the progress bar when contacts are loading.

**NOTE:**
There are 2 things I'm unsure of and would like some feed back
- Should I increase the number of results, or is 50 max results per query good?
- Is the progress bar useful. Maybe I didn't have enough contacts, but the loading was pretty fast for me. Tell me if it should be kept or removed if possible

### How to check PR:

1. Add many contacts to the Android device if you can
2. Run the app and login as any user if not already logged in
3. Click "Add Users"
4. Click "Invite from Contact List" (accept  permissions if requested)

You should see your contacts as the app always showed. 
Please also test the search bar.
